### PR TITLE
Add dropped alertmanagers to alertmanagers API 

### PR DESCRIPTION
**Pull request by Conorbro**
Following on from #3574

Alertmanagers dropped during relabelling will now be included in calls to `/api/v1/alertmanagers`

Example result:
```
{
  "status": "success",
  "data": {
    "activeAlertmanagers": [
      {
        "url": "http://127.0.0.1:9094/api/v1/alerts"
      }
    ],
    "droppedAlertmanagers": [
      {
        "url": "http://127.0.0.1:9093/api/v1/alerts"
      }
    ]
  }
}
```

@krasi-georgiev

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
@Conorbro thanks LGTM , but maybe @brian-brazil can find some nits.

File: notifier/notifier.go [hunk 1]
```diff
@@ -540,24 +560,28 @@ func newAlertmanagerSet(cfg *config.AlertmanagerConfig, logger log.Logger) (*ale
 // sync extracts a deduplicated set of Alertmanager endpoints from a list
 // of target groups definitions.
 func (s *alertmanagerSet) sync(tgs []*targetgroup.Group) {
-	all := []alertmanager{}
+	allAms := []alertmanager{}
+	allDams := []alertmanager{}
```
Comment (brian-brazil) [inline notifier/notifier.go]: It's not obvious from context what "Dams" are, the naming could be improved

Comment (Conorbro) [reply notifier/notifier.go]: Would "DroppedAms" work?

Comment (brian-brazil) [reply notifier/notifier.go]: Yes

**Comment by brian-brazil** [comment]
Do docs need updating?

**Comment by Conorbro** [comment]
@brian-brazil 

Docs have now been amended.

**Pull request merged**